src/couch_replicator_scheduler.erl

```diff
@@ -451,10 +453,19 @@ reschedule(State) ->
     ok.
 
 
+-spec stop_excess_jobs(#state{}, non_neg_integer()) -> ok.
 stop_excess_jobs(State, Running) ->
     #state{max_jobs=MaxJobs} = State,
-    if Running > MaxJobs ->
-        stop_jobs(Running - MaxJobs, State);
+    StopCount = Running - MaxJobs,
+    if StopCount > 0 ->
+        Stopped = stop_jobs(StopCount, true, State),
+        OneshotLeft = StopCount - Stopped,
+        if OneshotLeft > 0 ->
+            stop_jobs(OneshotLeft, false, State),
```

Comment:
I think it more clear how scheduler works if `max_jobs` really means `max_jobs`. This stopping code will only run after user (an operator) explicitly adjusts max_jobs to a lower value. Presumably they have some good reasons to do so. If this doesn't stop the jobs, there might be a need for a `really_max_jobs` setting in the future.

Also if we break the "Running <= MaxJobs"  constraint would mean changing the algorithm a bit as `rotate` function relies on it `if Running == MaxJobs, Pending > 0 -> .. start/stop ...`